hpat/pd_series_ext.py

```diff
@@ -469,6 +469,21 @@ def resolve_apply(self, ary, args, kws):
         # TODO: handle apply differences: extra args, np ufuncs etc.
         return self._resolve_map_func(ary, args, kws)
 
+    def _resolve_combine_func(self, ary, args, kws):
+        dtype = ary.dtype
+        # getitem returns Timestamp for dt_index and series(dt64)
+        if dtype == types.NPDatetime('ns'):
+            dtype = pandas_timestamp_type
+        code = args[1].value.code
+        f_ir = numba.ir_utils.get_ir_of_code({'np': np}, code)
+        f_typemap, f_return_type, f_calltypes = numba.compiler.type_inference_stage(
```

Comment:
This assumes that both series have the same `dtype` which is not necessarily the case.